### ISSUE TITLE
Support: asound config

### ISSUE THREAD

**Issue by bendikrb** [BlackLight/platypush#87]
This might be pretty far out of scope for this platform, but I thought I'd ask anyhow; I have platypush running on several rasberry pi's, together with an instance of [snapclient](https://github.com/badaix/snapcast).
I want platypush to lower the volume of the snapclient output upon `platypush.message.event.assistant.HotwordDetectedEvent`, and then set the volume to whatever it was before, at `platypush.message.event.assistant.ConversationEndEvent` (or whaterver).

I guess the core of my problem is; how to configure alsa so that I have a separate volume control for my snapclient? Do you by any chance have an example for this? :)

**Comment by BlackLight** [comment]
You can use the snapcast plugin to change the volume upon assistant interaction:

```yaml
event.hook.on_conversation_started:
    if:
        type: platypush.message.event.assistant.ConversationStartEvent
    then:
        - 
            action: music.snapcast.volume
            args:
                host: localhost
                client: hostname
                volume: 10

event.hook.on_conversation_ended:
    if:
        type: platypush.message.event.assistant.ConversationEndEvent
    then:
        - 
            action: music.snapcast.volume
            args:
                host: localhost
                client: hostname
                volume: 100
```

**Comment by bendikrb** [comment]
Awesome, that did the trick!
Except - I need another event than `platypush.message.event.assistant.ConversationEndEvent` to raise the volume. Now it raises the volume just before my `tts.google.say`.
Is the `then:` actions run sequentially? Or do you have another suggestion for what event I can listen to - for when the tts action has completed?

**Comment by BlackLight** [comment]
Then you could move the second `music.snapcast.volume` directive inside of your `ResponseEvent` hook, so it resets the volume only after the tts.

The actions in an event hook are indeed executed sequentially, so if you put `tts.google.say` before `music.snapcast.volume` the order should be respected. If you want to execute actions in parallel/asynchronously then you can wrap them in a `procedure.async.<name>` instead.

**Comment by bendikrb** [comment]
Perfect!
Any way to capture the current volume to a variable, and then use that in the `music.snapcast.volume` action later..?

**Comment by BlackLight** [comment]
You can use [`music.snapcast.status`] to query the status of a specific host/client and get the volume:

```yaml
# Getting and storing the volume in an in-memory variable

-
    action: music.snapcast.status
    args:
        host: localhost
        client: your-client

-
    action: variable.mset
    args:
        SNAPCAST_VOLUME: ${config['volume']['percent']}

# Restoring the volume

-
    action: variable.mget
    args:
        name: SNAPCAST_VOLUME

-
    action: music.snapcast.volume
    args:
        host: localhost
        client: your-client
        volume: ${SNAPCAST_VOLUME}
```
